src/hardware/pcspeaker.cpp

```diff
@@ -161,12 +166,12 @@ static void ForwardPIT(float newindex) {
 	}
 }
 
+// PIT-mode activation
 void PCSPEAKER_SetCounter(Bitu cntr,Bitu mode) {
-	if (!spkr.last_ticks) {
-		if(spkr.chan) spkr.chan->Enable(true);
-		spkr.last_index=0;
-	}
-	spkr.last_ticks=PIC_Ticks;
+	// Guard
+	if (!SpeakerExists())
+		return;
```

Comment:
The code in lines 172, 173 is more informative than comment, I think we can loose it ;)

Btw, let's format line 170 using clang-format, since we're touching a line before and a line after.

Reply:
Agreed :-), and gone.